float64 theta1
float64 theta2
float64 theta3
float64 theta4
float64 theta5
float64 theta6
float64 theta7
float64 finger1
float64 finger2